<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CalcDebugger</class>
 <widget class="QWidget" name="CalcDebugger">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>264</width>
    <height>671</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Debugger</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_4">
   <property name="sizeConstraint">
    <enum>QLayout::SetMinimumSize</enum>
   </property>
   <property name="margin">
    <number>0</number>
   </property>
   <item>
    <widget class="QComboBox" name="cb_target"/>
   </item>
   <item>
    <widget class="QToolBox" name="tbPages">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="pageBreakpoints">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>254</width>
        <height>499</height>
       </rect>
      </property>
      <attribute name="label">
       <string>Breakpoints</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="QListView" name="lw_breakpoints">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="gb_break_trigger">
         <property name="enabled">
          <bool>false</bool>
         </property>
         <property name="title">
          <string>Trigger</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <property name="margin">
           <number>2</number>
          </property>
          <item row="2" column="0">
           <widget class="QLabel" name="label_27">
            <property name="text">
             <string>Start</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1" colspan="2">
           <widget class="QLineEdit" name="le_break_start_addr">
            <property name="inputMask">
             <string>HHHHHHHH; </string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_28">
            <property name="text">
             <string>End</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1" colspan="2">
           <widget class="QLineEdit" name="le_break_end_addr">
            <property name="inputMask">
             <string>HHHHHHHH; </string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_29">
            <property name="text">
             <string>Mask</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="3">
           <widget class="QComboBox" name="cb_break_type">
            <item>
             <property name="text">
              <string>Memory read</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Memory execute</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Memory write</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Port read</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Port write</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Opcode</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Type</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="2">
           <widget class="QRadioButton" name="rb_break_logical">
            <property name="text">
             <string>Logical</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QRadioButton" name="rb_break_physical">
            <property name="text">
             <string>Physical</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1" colspan="2">
           <widget class="QLineEdit" name="le_break_mask_addr">
            <property name="inputMask">
             <string>HHHHHHHH; </string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="te_break_script">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pageCPU">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>254</width>
        <height>499</height>
       </rect>
      </property>
      <attribute name="label">
       <string>CPU</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <property name="margin">
        <number>0</number>
       </property>
       <item row="0" column="1">
        <widget class="QSpinBox" name="spn_refresh">
         <property name="suffix">
          <string> ms</string>
         </property>
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="value">
          <number>200</number>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Refresh</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QGroupBox" name="gbRegisters">
         <property name="title">
          <string/>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <property name="margin">
           <number>2</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="label_16">
            <property name="text">
             <string>af</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="CalcHexValue" name="le_af">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QLabel" name="label_20">
            <property name="text">
             <string>af'</string>
            </property>
           </widget>
          </item>
          <item row="0" column="4" colspan="2">
           <widget class="CalcHexValue" name="le_af_shadow">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_14">
            <property name="text">
             <string>bc</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="CalcHexValue" name="le_bc">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QLabel" name="label_18">
            <property name="text">
             <string>bc'</string>
            </property>
           </widget>
          </item>
          <item row="1" column="4" colspan="2">
           <widget class="CalcHexValue" name="le_bc_shadow">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_15">
            <property name="text">
             <string>de</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1" colspan="2">
           <widget class="CalcHexValue" name="le_de">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QLabel" name="label_19">
            <property name="text">
             <string>de'</string>
            </property>
           </widget>
          </item>
          <item row="2" column="4" colspan="2">
           <widget class="CalcHexValue" name="le_de_shadow">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_13">
            <property name="text">
             <string>hl</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1" colspan="2">
           <widget class="CalcHexValue" name="le_hl">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="3" column="3">
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>hl'</string>
            </property>
           </widget>
          </item>
          <item row="3" column="4" colspan="2">
           <widget class="CalcHexValue" name="le_hl_shadow">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="label_24">
            <property name="text">
             <string>ix</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1" colspan="2">
           <widget class="CalcHexValue" name="le_ix">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="4" column="3">
           <widget class="QLabel" name="label_21">
            <property name="text">
             <string>sp</string>
            </property>
           </widget>
          </item>
          <item row="4" column="4" colspan="2">
           <widget class="CalcHexValue" name="le_sp">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="label_23">
            <property name="text">
             <string>iy</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1" colspan="2">
           <widget class="CalcHexValue" name="le_iy">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="5" column="3">
           <widget class="QLabel" name="label_22">
            <property name="text">
             <string>pc</string>
            </property>
           </widget>
          </item>
          <item row="5" column="4" colspan="2">
           <widget class="CalcHexValue" name="le_pc">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="6" column="0">
           <widget class="QLabel" name="label_7">
            <property name="text">
             <string>ir</string>
            </property>
           </widget>
          </item>
          <item row="6" column="1" colspan="2">
           <widget class="CalcHexValue" name="le_ir">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HHHH; </string>
            </property>
           </widget>
          </item>
          <item row="6" column="3" colspan="3">
           <widget class="CalcFlags" name="lbl_flags">
            <property name="text">
             <string/>
            </property>
            <property name="textFormat">
             <enum>Qt::PlainText</enum>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QGroupBox" name="gb_disasm">
         <property name="title">
          <string/>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <property name="margin">
           <number>2</number>
          </property>
          <item row="0" column="0" colspan="3">
           <widget class="QLineEdit" name="le_disasm_start">
            <property name="inputMask">
             <string>HHHHHH; </string>
            </property>
           </widget>
          </item>
          <item row="1" column="0" colspan="5">
           <widget class="QListWidget" name="lv_disasm"/>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QPushButton" name="b_resume">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Resume</string>
            </property>
           </widget>
          </item>
          <item row="2" column="2" colspan="2">
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>68</width>
              <height>26</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="4">
           <widget class="QPushButton" name="b_step">
            <property name="enabled">
             <bool>false</bool>
            </property>
            <property name="text">
             <string>Step</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3" colspan="2">
           <widget class="QSpinBox" name="spn_disasm_length">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="suffix">
             <string> lines</string>
            </property>
            <property name="minimum">
             <number>5</number>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="value">
             <number>10</number>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pageMemory">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>254</width>
        <height>499</height>
       </rect>
      </property>
      <attribute name="label">
       <string>Memory</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Mapping</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QLabel" name="label_12">
            <property name="text">
             <string>Bank A</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="CalcHexValue" name="le_bankA">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HH; </string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QLabel" name="lbl_bankA">
            <property name="text">
             <string>Flash</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_25">
            <property name="text">
             <string>Bank B</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="CalcHexValue" name="le_bankB">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HH; </string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QLabel" name="lbl_bankB">
            <property name="text">
             <string>RAM</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_26">
            <property name="text">
             <string>Bank C</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="CalcHexValue" name="le_bankC">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>100</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="inputMask">
             <string>HH; </string>
            </property>
           </widget>
          </item>
          <item row="2" column="3">
           <widget class="QLabel" name="lbl_bankC">
            <property name="text">
             <string>RAM</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2" rowspan="3">
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="chk_flash_lock">
         <property name="text">
          <string>Flash locked</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_4">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="title">
          <string>HexView</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="pagePorts">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>254</width>
        <height>499</height>
       </rect>
      </property>
      <attribute name="label">
       <string>Ports</string>
      </attribute>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="0" margin="0"/>
 <customwidgets>
  <customwidget>
   <class>CalcHexValue</class>
   <extends>QLineEdit</extends>
   <header>calchexvalue.h</header>
  </customwidget>
  <customwidget>
   <class>CalcFlags</class>
   <extends>QLabel</extends>
   <header>calcflags.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
